src/raftstore/store/store.rs

```diff
@@ -1712,8 +1711,9 @@ impl<T: Transport, C: PdClient> Store<T, C> {
             //                  |-----------------threshold------------ |
             //              first_index                         replicated_index
             let replicated_idx = peer.raft_group
-                .status()
-                .progress
+                .raft
+                .prs()
+                .voters()
```

Comment:
we also need to handle learners here.

Reply:
sorry, i am not sure whether we need to handle learners here yet, i read the code before the learner feature been merged these days, but i make sure the the code(master branch now) did'n handle learners too before fire this PR.
agree with @BusyJay,  sounds more work must be do.

but anyway, if we need to handle learners here, somethink must be change here.

Reply:
we must handle learners later, the learner can also compact the raft log. So you can help us support this right now.

Reply:
i think if not include learner, compact_idx bigger than the matched index of some learner,
it maybe easy to  trigger the case where leader have to send snapshot to the learner